runner/full.go

```diff
@@ -89,13 +89,19 @@ func processServer(ctx context.Context, server *state.Server, globalCollectionOp
 	var collectionStatus state.CollectionStatus
 	var err error
 
+	if server.Pause.Pause == true {
+		logger.PrintWarning("Snapshot processing disabled by pganalyze server: %s", server.Pause.Reason)
+		return newState, newGrant, collectionStatus, nil
```

Comment:
My thought was that we don't want to preserve the existing state, since the diffs (e.g. for query stats) would become out of date while snapshot collection is paused.

This certainly could return an error instead of printing a warning. I'm curious what others think.

Reply:
I was thinking that this is somewhat close to the ignore replica case, where we handle it with an error. Though more I think about it, I think it's different and it's fine as is. I was also a bit worried about the consequences of storing the empty struct as a `server.PrevState` but that's how it is when you boot up the collector without a state file, so seems fine to me.